lib/plugins/creative.js

```diff
@@ -21,15 +21,38 @@ function inject (bot) {
   const creativeSlotsUpdates = []
 
   // WARN: This method should not be called twice on the same slot before first promise succeeds
-  async function setInventorySlot (slot, item) {
+  async function setInventorySlot (slot, item, waitTimeout = 400) {
     assert(slot >= 0 && slot <= 44)
 
     if (Item.equal(bot.inventory.slots[slot], item, true)) return
+
+    if (bot.registry.version['>=']('1.21.3')) {
+      bot._client.write('set_creative_slot', {
+        slot,
+        item: Item.toNotch(item)
+      })
+      // No ack
+      bot._setSlot(slot, item)
```

Comment:
the update slot event is emitted sync here; so the once update slot should be created before this function call

Reply:
ah nevermind you're waiting for a network update for the slot below